src/Library.php

```diff
@@ -126,22 +126,30 @@ public static function append_draft_order_post_status( $statuses ) {
 	}
 
 	/**
-	 * Delete draft orders older than a day.
+	 * Delete draft orders older than a day in batches of 20.
 	 *
 	 * Ran on a daily cron schedule.
 	 */
 	public static function delete_expired_draft_orders() {
-		global $wpdb;
-
-		$wpdb->query(
-			"
-			DELETE posts, term_relationships, postmeta
-			FROM $wpdb->posts posts
-			LEFT JOIN $wpdb->term_relationships term_relationships ON ( posts.ID = term_relationships.object_id )
-			LEFT JOIN $wpdb->postmeta postmeta ON ( posts.ID = postmeta.post_id )
-			WHERE posts.post_status = 'wc-checkout-draft'
-			AND posts.post_modified <= ( NOW() - INTERVAL 1 DAY )
-			"
+		$orders = wc_get_orders(
+			[
+				'date_modified' => '<' . strtotime( '-1 DAY' ),
+				'limit'         => 20,
+				'status'        => 'wc-checkout-draft',
+				'type'          => 'shop_order',
+			]
 		);
+		$count  = 0;
+
+		if ( $orders ) {
+			foreach ( $orders as $order ) {
+				$order->delete( false );
+				$count ++;
+			}
+		}
+
+		if ( 20 === $count && function_exists( 'as_enqueue_async_action' ) ) {
+			as_enqueue_async_action( 'woocommerce_cleanup_draft_orders' );
```

Comment:
Do we want this running as soon as possible, or should we stagger it a bit?

Reply:
This is equivalent to using using `time()` or `time() + seconds`. If I recall, AS will manage memory and resource limits but I'll double check.

Reply:
Mike shared this with me in slack which is a thing I learned about AS, so we should be okay here:

>  If there are actions to be processed, Action Scheduler will stake a unique claim for a batch of 25 actions and begin processing that batch. The PHP process spawned to run the batch will then continue processing batches of 25 actions until it uses 90% of available memory or has been processing for 30 seconds.